### PR TITLE
[actualmulti] iP

### PR DESCRIPTION
**Pull request by actualmulti**
# Jake
> "The more you learn, the less you know." -Mitchel Lee

Jake does anything you want it to do. It is,
* CLI-based
* So easy to learn and use

Steps to get it running: 
1. download it from [here](https://github.com/actualmulti/ip/releases/tag/A-Jar)
2. run the jar
3. profit 💰

It is 100% **FREE**

Features include:
- [x] todo list
- [x] deadline list
- [x] event list

Main method is as seen below: 
```java
    public static void main(String[] args) {
        new Jake("./data/jake.txt").run();
    }
```

### CODE REVIEW

File: src/main/java/jake/task/Task.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package jake.task;
+
+public abstract class Task {
```
Comment (CATISNOTSODIUM) [inline src/main/java/jake/task/Task.java:13]: Missing javadoc here (I know you don't have to add javadoc on all classes but this is just a code review)

File: src/main/java/jake/task/Task.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    protected final String name;
+    protected boolean done;
```
Comment (CATISNOTSODIUM) [inline src/main/java/jake/task/Task.java]: Should be `isDone` (following naming rules for boolean)

File: src/main/java/jake/Jake.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+ * Jake is a task management assistant.
+ * It supports todo tasks, deadline tasks, and event tasks with persistent storage.
+ *
+ * @author Mitchel lee
+ */
+public class Jake {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private GuiUi guiUi;
```
Comment (CATISNOTSODIUM) [inline src/main/java/jake/Jake.java:23]: I really like how you separate `Ui` and `GuiUi`.
Nice work 👍

File: data/jake.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+T | 0 | kill myself
```
Comment (CATISNOTSODIUM) [inline data/jake.txt]: You should remove this file.
I thought it's already been excluded in `.gitignore` file.

**Review by CATISNOTSODIUM — COMMENTED** [review]
A quick peer review

Overall really nice code! The structure is good and the testing is pretty comprehensive.

File: src/main/java/jake/Parser.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        } catch (Exception e) {
+            throw new JakeException("Deadline task must have a valid name and/or date!");
+        }
+    }
+
+    /**
+     * Parses an event command to extract the task name, start date, and end date.
+     * Expected format: "event [task name] /[start date] /[end date]"
+     *
+     * @param fullCommand the complete event command string
```
Comment (dlywlotus) [inline src/main/java/jake/Parser.java:234]: I like that the java docs are descriptive.

File: src/test/java/jake/ParserTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
+package jake;
```
Comment (dlywlotus) [inline src/test/java/jake/ParserTest.java:1]: Its great that the tests are comprehensive and accounts for many edge cases.

File: src/test/java/jake/ui/UiTest.java [hunk 7]
```diff
@@ -0,0 +1,238 @@
+package jake.ui;
+
```
Comment (dlywlotus) [inline src/test/java/jake/ui/UiTest.java:2]: Once again, good testing here.

**Review by dlywlotus — COMMENTED** [review]
Overall, the code is well structured and well tested. LGTM! 😊
